dsgrid/config/annual_time_dimension_config.py

```diff
@@ -70,73 +72,22 @@ def convert_dataframe(
         self,
         df,
         project_time_dim,
-        model_years=None,
-        value_columns=None,
+        value_columns: set[str],
+        scratch_dir_context: ScratchDirContext,
         wrap_time_allowed=False,
         data_adjustment=None,
     ):
-        assert model_years is not None
-        assert value_columns is not None
-        match self.model.measurement_type:
-            case MeasurementType.MEASURED:
-                df = self.map_annual_time_measured_to_datetime(df, project_time_dim, model_years)
-            case MeasurementType.TOTAL:
-                df = self.map_annual_total_to_datetime(
-                    df, project_time_dim, model_years, value_columns
-                )
-            case _:
-                raise NotImplementedError(f"Unhandled: {self.model.measurement_type}")
-
-        return df
-
-    def map_annual_time_measured_to_datetime(self, annual_df, dt_dim, model_years):
-        """Map a dataframe with MeasuredType.MEASURED to DateTime."""
-        time_col = self.get_load_data_time_columns()[0]
-        df = dt_dim.build_time_dataframe(model_years=model_years)
-        dt_time_col = dt_dim.get_load_data_time_columns()[0]
-        return df.join(annual_df, on=F.year(dt_time_col) == annual_df[time_col]).drop(time_col)
-
-    def map_annual_total_to_datetime(self, annual_df, dt_dim, model_years, value_columns):
-        """Map a dataframe with MeasuredType.TOTAL to DateTime."""
-        assert (
-            dt_dim.model.time_type == TimeDimensionType.DATETIME
-        ), "dt_dim must be datetime type."
-        frequency = dt_dim.model.frequency
-        time_col = self.get_load_data_time_columns()[0]
-        dfs = []
-        for model_year in model_years:
-            if self.model.include_leap_day and model_year % 4 == 0:
-                measured_duration = timedelta(days=366)
-            else:
-                measured_duration = timedelta(days=365)
-            df = dt_dim.build_time_dataframe(model_years=[model_year])
-            dt_time_col = dt_dim.get_load_data_time_columns()[0]
-            with custom_spark_conf({"spark.sql.session.timeZone": "UTC"}):
-                tmp_col = f"{dt_time_col}_utc"
-                df = df.withColumn(
-                    tmp_col,
-                    F.from_utc_timestamp(
-                        dt_time_col, dt_dim.model.datetime_format.timezone.tz_name
-                    ),
-                )
-                df = df.join(annual_df, on=F.year(df[tmp_col]) == annual_df[time_col]).drop(
-                    time_col, tmp_col
-                )
-            for column in value_columns:
-                df = df.withColumn(column, F.col(column) / (measured_duration / frequency))
-            if not df.rdd.isEmpty():
-                dfs.append(df)
-
-        return union(dfs)
+        # Currently, only map_annual_historical_time_to_date_time is supported.
```

Comment:
This change removes polymorphism for this `convert_dataframe` interface. It is difficult, if not impossible, to make one consistent interface because mapping time dimensions from dataset to project may require other dimensions.